### PR TITLE
Add Maps Stats API documentation

### LINKED ISSUE #3711: [DOC] Maps: add stats API for maps and layers in maps plugin
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
For 2.7.0 maps plugin  release, we will add the OSD Rest API `/api/maps-dashboards/stats` to get stats for maps and layers, which need to be added on doc website. https://opensearch.org/docs/2.6/dashboards/visualize/maps

* GH issue: https://github.com/opensearch-project/dashboards-maps/pull/362
* GH PR: https://github.com/opensearch-project/dashboards-maps/issues/363
* API example:
```
{
   "maps_total":4,
   "layers_filters_total":4,
   "layers_total":{
      "opensearch_vector_tile_map":2,
      "documents":7,
      "wms":1,
      "tms":2
   },
   "maps_list":[
      {
         "id":"88a24e6c-0216-4f76-8bc7-c8db6c8705da",
         "layers_filters_total":4,
         "layers_total":{
            "opensearch_vector_tile_map":1,
            "documents":3,
            "wms":0,
            "tms":0
         }
      },
      {
         "id":"4ce3fe50-d309-11ed-a958-770756e00bcd",
         "layers_filters_total":0,
         "layers_total":{
            "opensearch_vector_tile_map":0,
            "documents":2,
            "wms":0,
            "tms":1
         }
      },
      {
         "id":"af5d3b90-d30a-11ed-a605-f7ad7bc98642",
         "layers_filters_total":0,
         "layers_total":{
            "opensearch_vector_tile_map":1,
            "documents":1,
            "wms":0,
            "tms":1
         }
      },
      {
         "id":"5ca1ec10-d30b-11ed-a042-93d8ff0f09ee",
         "layers_filters_total":0,
         "layers_total":{
            "opensearch_vector_tile_map":0,
            "documents":1,
            "wms":1,
            "tms":0
         }
      }
   ]
}
```

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Fixes #3711

### Checklist
- [x ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _dashboards/visualize/maps-stats-api.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+grand_parent: Building data visualizations
+parent: Using coordinate and region maps 
+has_children: false
+---
+
+# Maps Stats API
+Introduced 2.7
+{: .label .label-purple }
+
+When you create a [map]({{site.url}}{{site.baseurl}}/dashboards/visualize/maps/) in OpenSearch Dashboards, the map is registered as a saved object in the [Maps plugin](https://github.com/opensearch-project/dashboards-maps). The Maps Stats API provides information about all maps registered in the Maps plugin.
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _dashboards/visualize/maps-stats-api.md]: Reworded.

File: _dashboards/visualize/maps-stats-api.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+{: .label .label-purple }
+
+When you create a [map]({{site.url}}{{site.baseurl}}/dashboards/visualize/maps/) in OpenSearch Dashboards, the map is registered as a saved object in the [Maps plugin](https://github.com/opensearch-project/dashboards-maps). The Maps Stats API provides information about all maps registered in the Maps plugin. 
+
+#### Example request
+
+You can access the Maps Stats API by providing its address in the following format:
+
+```
+<opensearch-dashboards-endpoint-address>:<port>/api/maps-dashboards/stats
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _dashboards/visualize/maps-stats-api.md:21]: @junqiu-lei In the example, we have a port `http://localhost:5601/api/maps-dashboards/stats`. Is the port always/sometimes/never needed?

File: _dashboards/visualize/maps-stats-api.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+The following table lists all response fields.
+
+| Field | Data type | Description |
+| :--- | :--- | :--- | 
+| `maps_total` | Integer | The total number of maps registered as saved objects with the Maps plugin. |
+| `layers_filters_total` | Integer | The total number of filters for all layers in all maps. This includes [layer-level filters]({{site.url}}{{site.baseurl}}/dashboards/visualize/maps/#filtering-data-at-the-layer-level) but excludes global filters like [shape filters]({{site.url}}{{site.baseurl}}/dashboards/visualize/maps/#drawing-shapes-to-filter-data). |
+| `layers_total` | Object | Totals statistics for all layers in all maps. |
+| `layers_total.opensearch_vector_tile_map` | Integer | The total number of basemaps in all maps. |
+| `layers_total.documents` | Integer | The total number of documents in all layers of all maps. |
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _dashboards/visualize/maps-stats-api.md:115]: @junqiu-lei Is `layers_total.documents` the number of document layers, not the number of individual documents in all layers?

[the remark this answers is not in the document]
Comment (junqiu-lei) [reply _dashboards/visualize/maps-stats-api.md:115]: Yes

**Review by junqiu-lei — APPROVED** [review]
LGTM, thank you!

**Review by ariamarble — APPROVED** [review]
Looks good. I don't have any additional feedback. Thanks.

File: _dashboards/visualize/maps-stats-api.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+```
+
+The OpenSearch Dashboards endpoint address may contain a port number if it is specified in the OpenSearch configuration file. The specific URL format depends on the type of OpenSearch deployment and the network environment in which it is hosted.
+{: .note}  
+
+You can query the endpoint in two ways:
+  
+  - By accessing the endpoint address (for example`http://localhost:5601/api/maps-dashboards/stats`) in a browser
+
+  - By using the `curl` command in the terminal:
```
Comment (vagimeli) [inline _dashboards/visualize/maps-stats-api.md:31]: Should there be a line space between lines 31 and 32?

File: _dashboards/visualize/maps-stats-api.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+      }
+   ]
+}
+```
+
+## Response fields
+
+The response contains statistics for the following layer types:
+
+- Basemaps: Either an OpenSearch default map or custom base layer maps you created.
```
Comment (vagimeli) [inline _dashboards/visualize/maps-stats-api.md:100]:
```suggestion
- Basemaps: Either an OpenSearch default map or custom base layer map you created.
```

File: _dashboards/visualize/maps-stats-api.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+- WMS layers: Custom WMS base layer maps you created.
+
+- TMS layers: Custom TMS base layer maps you created.
+
+- Document layers: The data layers of the map.
```
Comment (vagimeli) [inline _dashboards/visualize/maps-stats-api.md:106]:
```suggestion
- Document layers: The map's data layers.
```

File: _dashboards/visualize/maps-stats-api.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+| `layers_total` | Object | Totals statistics for all layers in this map. |
+| `layers_total.opensearch_vector_tile_map` | Integer | The total number of OpenSearch basemaps in this map. |
+| `layers_total.documents` | Integer | The total number of document layers in this map. |
+| `layers_total.wms` | Integer | The total number of WMS layers in this map. |
+| `layers_total.tms` | Integer | The total number of TMS layers in this map. |
+
+The saved object ID helps you navigate to a particular map because the ID is the last part of the map's URL. For example, in OpenSearch Playground, the address of the `[Flights] Flights Status on Maps Destination Location` map is `https://playground.opensearch.org/app/maps-dashboards/88a24e6c-0216-4f76-8bc7-c8db6c8705da`, where `88a24e6c-0216-4f76-8bc7-c8db6c8705da` is the saved object ID for this map.
```
Comment (vagimeli) [inline _dashboards/visualize/maps-stats-api.md:135]: Great tip! I like how you gave this example.

**Review by vagimeli — APPROVED** [review]

File: _dashboards/visualize/maps-stats-api.md (hunk 5 above)
Comment (kolchfa-aws) [reply _dashboards/visualize/maps-stats-api.md:100]: There may be more than one base layer map, so I'll keep plural here :)

File: _dashboards/visualize/maps-stats-api.md (hunk 4 above)
Comment (kolchfa-aws) [reply _dashboards/visualize/maps-stats-api.md:31]: No, this renders with correct spacing automatically.

File: _dashboards/visualize/maps-stats-api.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+| `layers_total.tms` | Integer | The total number of TMS layers in all maps. |
+| `maps_list` | Array | A list of all maps saved in OpenSearch Dashboards. |
+
+Each map in the `map_list` contains the following fields.
+
+| Field | Data type | Description |
+| :--- | :--- | :--- | 
+| `id` | String | The map's saved object ID. |
+| `layers_filters_total` | Integer | The total number of filters for all layers in the map. This includes [layer-level filters]({{site.url}}{{site.baseurl}}/dashboards/visualize/maps/#filtering-data-at-the-layer-level) but excludes global filters like [shape filters]({{site.url}}{{site.baseurl}}/dashboards/visualize/maps/#drawing-shapes-to-filter-data) . |
+| `layers_total` | Object | Totals statistics for all layers in this map. |
```
Comment (natebower) [inline _dashboards/visualize/maps-stats-api.md:129]:
```suggestion
| `layers_total` | Object | Totals statistics for all layers in the map. |
```

File: _dashboards/visualize/maps-stats-api.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+| `layers_total.opensearch_vector_tile_map` | Integer | The total number of OpenSearch basemaps in this map. |
+| `layers_total.documents` | Integer | The total number of document layers in this map. |
+| `layers_total.wms` | Integer | The total number of WMS layers in this map. |
```
Comment (natebower) [inline _dashboards/visualize/maps-stats-api.md:132]:
```suggestion
| `layers_total.wms` | Integer | The total number of WMS layers in the map. |
```

File: _dashboards/visualize/maps-stats-api.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+| `layers_total.tms` | Integer | The total number of TMS layers in this map. |
```
Comment (natebower) [inline _dashboards/visualize/maps-stats-api.md:133]:
```suggestion
| `layers_total.tms` | Integer | The total number of TMS layers in the map. |
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

**Pull request merged**